api/types.go

```diff
@@ -12,21 +15,31 @@ type ProxyConfig struct {
 }
 
 type OIDCConfig struct {
```

Comment:
I would prefer having the attributes of the file in its own type to separate the deserializing from the other type.